### ISSUE TITLE
"not enough memory" error from itp_parent()/error()

### ISSUE THREAD

**Issue by jamessan** [neovim/neovim#9212]
- `nvim --version`: v0.3.1
- Operating system/version: Debian/Ubuntu

The unit tests, particularly mbyte_spec.lua's "utf_char2bytes" test, are prone to [failing](https://launchpadlibrarian.net/396305149/buildlog_ubuntu-bionic-amd64.neovim_0.3.1-1ppa1~ubuntu18.04.1_BUILDING.txt.gz) with a "not enough memory" error.  I can somewhat reliably reproduce the behavior if I have something CPU intensive running when I run that unit test.

```
[ RUN      ] mbyte utf_char2bytes: FAIL
test/unit/helpers.lua:731: (string) '
not enough memory'

stack traceback:
	test/unit/helpers.lua:747: in function 'itp_parent'
	test/unit/helpers.lua:774: in function <test/unit/helpers.lua:764>
```

This has been one of the main problems with the Debian builds, as well as providing the stable PPA builds.

At a guess, we're somehow getting into a recursive error situation.

Cc @ZyX-I

**Comment by jamessan** [comment]
I hit this again and was able to strace the unit test while it was running.  [luajit.txt](https://github.com/neovim/neovim/files/2568887/luajit.txt) is a small snapshot of what what it was doing.  Watching it in top, it _is_ just chewing up memory.

**Comment by jamessan** [comment]
Fixed by 25356f2802b5b98efe7f0d6661979b0a919c4d2d
